python/ray/tune/syncer.py

```diff
@@ -758,6 +832,8 @@ def on_checkpoint(
             )
 
     def wait_for_all(self):
+        self._cleanup_trial_sync_processes()
```

Comment:
I'm new to this code; can you help me understand why we create trial sync processes and don't use Ray task/actor primitives? Maybe we do but I just haven't seen :)